tests/test_data_utils/test_append_observation.py

```diff
@@ -1,96 +1,58 @@
 import pandas as pd
 import unittest
-from spac.data_utils import append_observation
+from spac.data_utils import append_annotation
 
 
-class TestAppendObservation(unittest.TestCase):
+class TestAppendAnnotation(unittest.TestCase):
 
-    def setUp(self):
-        # Create a sample DataFrame for testing
-        self.data = pd.DataFrame({
-            'A': [1, 2, 3, 4],
-            'B': ['apple', 'banana', 'cherry', 'date']
-        })
+    def test_valid_annotation(self):
+        data = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
+        annotation = [{'C': 'Alice'}, {'D': 30}]
+        result = append_annotation(data.copy(), annotation)
+        self.assertTrue(all(col in result.columns for col in ['C', 'D']))

```

Comment:
@ruiheesi , would you please also assert that the value of these columns in the constant you provided "Alice" and 30?